### PR TITLE
`azurerm_palo_alto_next_generation_firewall_*` - support the property `trustedRanges`

### PR DESCRIPTION
**Pull request by neil-yechenwei**
Service team is requesting this new feature.

image

Note: Above test cases are also failed in Teamcity Daily Run with same error so that it's not related with this PR.

### CODE REVIEW

**Review by catriona-m — CHANGES_REQUESTED** [review]
Hi @neil-yechenwei - it looks like the test has a fail related to this PR in the latest run:

```
------- Stdout: -------
=== RUN   TestAccPaloAltoNextGenerationFirewallLocalRulestackVNet_update
=== PAUSE TestAccPaloAltoNextGenerationFirewallLocalRulestackVNet_update
=== CONT  TestAccPaloAltoNextGenerationFirewallLocalRulestackVNet_update
    testcase.go:113: Step 3/8 error: Error running pre-apply refresh: exit status 1
        Error: expected "network_profile.0.trusted_ranges.0" to be a valid CIDR Value, got 20.22.92.11: invalid CIDR address: 20.22.92.11
          with azurerm_palo_alto_next_generation_firewall_virtual_network_local_rulestack.test,
          on terraform_plugin_test.tf line 138, in resource "azurerm_palo_alto_next_generation_firewall_virtual_network_local_rulestack" "test":
         138:     trusted_ranges            = ["20.22.92.11", "20.23.92.11"]
        Error: expected network_profile.0.trusted_ranges.0 to contain a valid IP range, got: 20.22.92.11
          with azurerm_palo_alto_next_generation_firewall_virtual_network_local_rulestack.test,
          on terraform_plugin_test.tf line 138, in resource "azurerm_palo_alto_next_generation_firewall_virtual_network_local_rulestack" "test":
         138:     trusted_ranges            = ["20.22.92.11", "20.23.92.11"]
        Error: expected "network_profile.0.trusted_ranges.1" to be a valid CIDR Value, got 20.23.92.11: invalid CIDR address: 20.23.92.11
          with azurerm_palo_alto_next_generation_firewall_virtual_network_local_rulestack.test,
          on terraform_plugin_test.tf line 138, in resource "azurerm_palo_alto_next_generation_firewall_virtual_network_local_rulestack" "test":
         138:     trusted_ranges            = ["20.22.92.11", "20.23.92.11"]
        Error: expected network_profile.0.trusted_ranges.1 to contain a valid IP range, got: 20.23.92.11
          with azurerm_palo_alto_next_generation_firewall_virtual_network_local_rulestack.test,
          on terraform_plugin_test.tf line 138, in resource "azurerm_palo_alto_next_generation_firewall_virtual_network_local_rulestack" "test":
         138:     trusted_ranges            = ["20.22.92.11", "20.23.92.11"]
--- FAIL: TestAccPaloAltoNextGenerationFirewallLocalRulestackVNet_update (2307.17s)
FAIL
```

**Comment by neil-yechenwei** [comment]
@catriona-m , thanks for the comment. I updated PR. Please take another look. Thanks.

File: internal/services/paloalto/schema/network_profile.go [hunk 1]
```diff
@@ -67,6 +70,18 @@ func VnetNetworkProfileSchema() *pluginsdk.Schema {
 					},
 				},
 
+				"trusted_ranges": {
```
Comment (katbyte) [inline internal/services/paloalto/schema/network_profile.go:73]: should this be
```suggestion
				"trusted_addresses_ranges": {
```
or
```suggestion
				"trusted_addresses": {
```
to both be more clear what it is looking for as well match other propertyes int he provider?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/paloalto/schema/network_profile.go (hunk 1 above)
Comment (neil-yechenwei) [reply internal/services/paloalto/schema/network_profile.go]: Updated

**Comment by neil-yechenwei** [comment]
@katbyte , thanks for the comments. I updated PR. Please take another look. Thanks.

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @neil-yechenwei - checked tests and they are all failing?

```
------- Stdout: -------
=== RUN   TestAccPaloAltoNextGenerationFirewallVHubLocalRulestack_basic
=== PAUSE TestAccPaloAltoNextGenerationFirewallVHubLocalRulestack_basic
=== CONT  TestAccPaloAltoNextGenerationFirewallVHubLocalRulestack_basic
    testcase.go:113: Step 1/2 error: Error running apply: exit status 1
        
        Error: waiting for Virtual Hub: (Name "acctestVHUB-240118040419856605" / Resource Group "acctestRG-PANGFWVH-240118040419856605") provisioning route: failed to provision routing on Virtual Hub "acctestRG-PANGFWVH-240118040419856605" (Resource Group "acctestVHUB-240118040419856605")
        
          with azurerm_virtual_hub.test,
          on terraform_plugin_test.tf line 49, in resource "azurerm_virtual_hub" "test":
          49: resource "azurerm_virtual_hub" "test" {
        
    testing_new.go:90: Error running post-test destroy, there may be dangling resources: exit status 1
        
        Error: deleting Virtual Wan: (Name "acctestvwan-240118040419856605" / Resource Group "acctestRG-PANGFWVH-240118040419856605"): network.VirtualWansClient#Delete: Failure sending request: StatusCode=400 -- Original Error: Code="InUseVirtualWanCannotBeDeleted" Message="Cannot proceed with the operation because Virtual WAN has references. The WAN is referenced by 1 Virtual Hubs and 0 VPN Sites." Details=[]
        
--- FAIL: TestAccPaloAltoNextGenerationFirewallVHubLocalRulestack_basic (473.16s)
FAIL
```
could we plase fix these up so we can ensure this works as intended?

**Comment by neil-yechenwei** [comment]
Hi @katbyte , the failed test cases are also failed in both [Network](https://hashicorp.teamcity.com/buildConfiguration/TF_AzureRM_AZURERM_SERVICE_PUBLIC_NETWORK/87037?buildTab=tests&status=failed&name=virtualhub&expandedTest=build%3A%28id%3A87037%29%2Cid%3A2000000362) and [PaloAlto](https://hashicorp.teamcity.com/buildConfiguration/TF_AzureRM_AZURERM_SERVICE_PUBLIC_PALOALTO/87547?hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildTestsSection=true&expandBuildDeploymentsSection=false&expandBuildChangesSection=true) in Teamcity Daily Run with same error. Per the error message returned by API in teamcity, it's service API issue not TF issue. It depends on the fix from service team, so I can't fix it in the short term. I suggest to review/merge this PR first since the failure is not related with this PR. In the meanwhile, I have already filed azure support ticket and IcM ticket to ask service team to fix it. Does it make sense?

Note: As our test service principal and test subscription have insufficient permission for this RP, so it's hard to us to run some test cases on my local for this RP. I assume jackofallops has full permission for this RP since he added them in TF.

**Comment by katbyte** [comment]
@neil-yechenwei - is the problem causing the tests to fail described anywhere? and is there an ETA for the fix?

**Comment by neil-yechenwei** [comment]
@katbyte , for Virtual Hub issue, Virtual Wan Service team confirmed that the issue has been fixed last Friday and I found the Virtual Hub issue has gone in the latest Teamcity Daily Run.

**Comment by neil-yechenwei** [comment]
Below is the latest test result.

image

**Review by katbyte — APPROVED** [review]
Great thanks @neil-yechenwei - LGTM 🪟

**Pull request merged**